lockfile.go

```diff
@@ -56,6 +57,10 @@ func GetLockfile(path string) (Locker, error) {
 	if err != nil {
 		return nil, err
 	}
+	_, _, err = syscall.Syscall(syscall.SYS_FCNTL, uintptr(fd), syscall.F_SETFD, 0)
+	if err != syscall.Errno(0x0) {
+		return nil, err
+	}
```

Comment:
We don't get as much in the way of diagnostic feedback from it, but it looks like `unix.CloseOnExec()` is meant to be used for this.